core/net/ip/tcpip.c

```diff
@@ -230,8 +243,7 @@ tcp_connect(const uip_ipaddr_t *ripaddr, uint16_t port, void *appstate)
     return NULL;
   }
 
-  c->appstate.p = PROCESS_CURRENT();
-  c->appstate.state = appstate;
+  setup_appstate(&c->appstate, init_appstate);
```

Comment:
oops sorry for being unclear, the naming suggestion `init_appstate` was for the function name (replaces `setup_appstate`), not for the parameter.